docs/developers/react-development-guidelines.md

```diff
@@ -0,0 +1,397 @@
+if (x === 'rejected' && y === 4) { 
+  // do stuff 
+} 
+// ✅ DO
+enum PermissionEnum { // all enums should be in PascalCase and suffixed with "Enum" 
+  rejected = "rejected" 
+} 
+const ACCESS_LEVEL = 4; // all constants declared on top of functions should be in UPPER_CASE
+if (x === PermissionsEnum.rejected && y === ACCESS_LEVEL)
+{ 
+  //do stuff 
+}
+```
+
+* No **inline functions** in TSX.
+
+```jsx
+// 🚫 DON'T 
+<TouchableOpacity onPress={() => setPressed(true)}/> 
+// ✅ DO 
+const handlePress = () => { 
+  setPressed(true) 
+} 
+<TouchableOpacity onPress={handlePress}/>
+```
+
+* Use implicit `true` for **boolean** props
+
+```jsx
+// 🚫 DON'T 
+<Card isFullscreen={true}/> 
+// ✅ DO 
+<Card isFullscreen /> 
+```
+
+* No **`useCallback` or `useMemo` or `React.memo` unless really necessary**. Since the release of hooks, over-optimization has become a big problem.
+
+```jsx
+// 🚫 DON'T 
+const handlePress = useCallback(() => setPressed(true)); 
+<TouchableOpacity onPress={handlePress}/> 
+// 🚫 DON'T 
+const value = useMemo(() => user.level * multiplicator); 
+
+// ✅ DO 
+const handlePress = useCallback(() => setPressed(true)); 
+<Context.Provider value={{onPress: handlePress}}> 
+  {children} 
+</Context.Provider>
+```
+
+_**Rules of hooks:**_
+
+1. **Fake modularization**:
+  * Custom hooks may give the impression of modularization, but their logic runs inline in the parent component.
+  * State changes and reactive behavior in the hook will cause a rerender of the host component and any other hooks that depend on the updated hook.
+2. **Hook return values**:
+  * Custom hooks should return either a single function (for a lazy hook) or an object containing a function and some properties;
+  * Avoid returning objects with multiple functions to ensure consistency and maintainability;
+  * Use interfaces for hook return type:
+    ```jsx
+    useMyHook({firstParam, secondParam}: UseMyHookParamsType): UseMyHookReturnType => {}
+    ```
+3. **State management and data fetching**:
+  * Lifting state up should be used sparingly; hooks should primarily be used to gather state in one container and distribute it to child components.
+  * Strive to couple data fetching with UI rendering as isolating as possible to prevent unnecessary rerenders.
+  * Ensure one hook does not trigger the rerender of another by carefully managing dependencies and side effects.
+  * If only 10% of the lines of code in a hook do specific React logic like state management, or calling a selector from Redux, consider splitting the hook into:
+    - a smaller hook that does the React logic and returns the state, and
+    - a statless function that will be called inside the hook and will do the rest of the logic. This stateless function should be tested separately.
+4. **Hook interfaces**:
+  * Use interfaces for hook params if you're passing more than one argument to the hook invocation:
+    ```jsx
+    useMyHook({firstParam, secondParam}: UseMyHookParamsType) => {}
```

Comment:
Sometimes is simpler to see the props type inline using inline declaration. But it becomes ugly just in case you have a huge object. I assume for 3-4 fields is not necessary to extract a type